src/osmet.rs

```diff
@@ -0,0 +1,969 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::ffi::OsStr;
+use std::fs::{File, OpenOptions};
+use std::io::{self, BufReader, BufWriter, ErrorKind, Read, Seek, SeekFrom, Write};
+use std::os::unix::ffi::{OsStrExt, OsStringExt};
+use std::os::unix::fs::FileTypeExt;
+use std::os::unix::io::AsRawFd;
+use std::path::{Path, PathBuf};
+
+use error_chain::bail;
+use nix::mount;
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
```

Comment:
I think we should at least consider a build-time option to use openssl.  I doubt the `sha2` crate is FIPS certifed for example.

Reply:
Hmm, hadn't considered this. Yes, will investigate. (One thing I noticed is that `sha2` takes much longer to hash than `sha256sum`. Maybe using openssl would help with this.)

Reply:
(Depending on openssl on the other hand makes the binary much less portable)

Reply:
Hmm, though do we support running coreos-installer as a fat binary? I think the only supported paths are in the live ISO/PXE or the container (or build it yourself of course). (/cc @bgilbert) E.g. we already link to liblzma right now for example.

Reply:
OK nice, not surprisingly OpenSSL is at least as fast as `sha256sum`.